distribution/docker/druid.sh

```diff
@@ -45,45 +45,56 @@ echo "$(date -Is) startup service $SERVICE"
 
 # We put all the config in /tmp/conf to allow for a
 # read-only root filesystem
-cp -r /opt/druid/conf /tmp/conf
+mkdir -p /tmp/conf/
+cp -r /opt/druid/conf/druid /tmp/conf/druid
+
+getConfPath() {
+    cluster_conf_base=/tmp/conf/druid/cluster
+    case "$1" in
+    _common) echo $cluster_conf_base/_common ;;
+    historical) echo $cluster_conf_base/data/historical ;;
+    middleManager) echo $cluster_conf_base/data/middleManager ;;
+    coordinator | overlord) echo $cluster_conf_base/master/coordinator-overlord ;;
```

Comment:
do we still want to support users that want to run overlord and coordinator separately? They wouldn't be able to do so with a single config layout.

Reply:
it is still possible I think, one container would be running one of the following druid process...
- standalone coordinator process (in that case runtime.properties could contain configs relevant to coordinator)
- standalone overlord process (in that case runtime.properties could contain confits relevant to overlord)
- combined coordinator+overlord process (in that case runtime.properties could contain all config)

Also, it looks like possible to use "DRUID_CONFIG_COMMON" and "DRUID_CONFIG_{service}" variables to use a different directory layout. (inferring from line #114 on changed file)

Disclaimer: I haven't really done any of above, these are observations made by quickly reading the script :)



I think current state is trying to make it super easy for druid example cluster config to work with minimal/no changes which is a worth goal for people trying things out and setting up a small cluster.

That said, It might still be a good idea to support separate overlord/coordinator process config in a more obvious way or at least document how to achieve that.

Reply:
@ilhanadiyaman please confirm if you think above makes or does not make sense.

Reply:
@himanshug thanks for detailed explanations. Yes, indeed the reason is to make the setup as fast as possible with minimum configuration. That's why we use the example configuration file from [coordinator and overlord](https://github.com/apache/incubator-druid/blob/master/examples/conf/druid/cluster/master/coordinator-overlord/runtime.properties).

I think it is possible to run standalone coordinator process and standalone overlord process with this setup. You can check the [docker-compose.yml](https://github.com/apache/incubator-druid/blob/master/distribution/docker/docker-compose.yml). It's running both of them separately. Moreover, you can pass the full path of the configuration file per service as an environment variable.

`DRUID_CONFIG_${service} -- full path to a file for druid 'service' properties`

That said, because of the line `druid.coordinator.asOverlord.enabled=true` in the example config file, coordinator process might run Overlord process as well. I didn't check if there're any consequences of this setup but so far I didn't have any problems. Maybe we shouldn't use the example configuration files at all. But then it will put some learning process for people to understand and create the configuration files by themselves.

What do you think? Any ides that we can improve the process?